java-client/session/src/main/java/io/deephaven/client/impl/DataAndTypedTickets.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.client.impl;
+
+import com.google.protobuf.ByteString;
+import io.deephaven.client.impl.ObjectService.Bidirectional;
+import io.deephaven.client.impl.ObjectService.MessageStream;
+import io.deephaven.proto.backplane.grpc.Data;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Data and typed tickets to be sent from the client to the server as part of a
+ * {@link Bidirectional#messageStream(MessageStream) bidirection message stream}.
+ */
+public final class DataAndTypedTickets {
```

Comment:
Maybe we want to make the proto asymmetric, too?
Same kind of quibble regarding this class name. ClientMessage? MessageToServer?

Reply:
I agree - I think it makes sense to have two separate proto objects so they can be documented independently.

Reply:
If we're renaming the proto types, should be ClientData/ServerData, or we're renaming the message oneof (or we end up with foo.message.message, which we wanted to avoid).